gemrb/core/DisplayMessage.h

```diff
@@ -63,46 +57,49 @@ class GEM_EXPORT DisplayMessage
-	/** returns the speaker's color and name */
-	Color GetSpeakerColor(String& name, const Scriptable *&speaker) const;
 	/** displays any string in the textarea */
-	void DisplayMarkupString(String txt) const;
+	void DisplayMarkupString(String txt) const;	
 	/** displays a string constant in the textarea */
-	void DisplayConstantString(size_t stridx, const Color &color, Scriptable *speaker = nullptr) const;
-	/** displays actor name - action : parameter */
-	void DisplayConstantStringNameString(size_t stridx, const Color &color, size_t stridx2, const Scriptable *actor) const;
+	void DisplayConstantString(size_t stridx, unsigned char color, Scriptable *speaker = nullptr) const;
 	/** displays a string constant followed by a number in the textarea */
-	void DisplayConstantStringValue(size_t stridx, const Color &color, ieDword value) const;
-	/** displays a string constant in the textarea, starting with speaker's name */
+	void DisplayConstantStringValue(size_t stridx, unsigned char color, ieDword value) const;	
+	/** displays actor name - action : parameter */
+	void DisplayConstantStringName(size_t stridx, unsigned char color, const Scriptable *speaker) const;
 	void DisplayConstantStringName(size_t stridx, const Color &color, const Scriptable *speaker) const;
-	/** displays a string constant in the textarea, starting with speaker's name, also replaces one numeric value (it is a format string) */
-	void DisplayConstantStringNameValue(size_t stridx, const Color &color, const Scriptable *speaker, int value) const;
 	/** displays a string constant in the textarea, starting with actor, and ending with target */
-	void DisplayConstantStringAction(size_t stridx, const Color &color, const Scriptable *actor, const Scriptable *target) const;
+	void DisplayConstantStringAction(size_t stridx, unsigned char color, const Scriptable *actor, const Scriptable *target) const;	
+	/** displays a string in the textarea, starting with speaker's name */
+	void DisplayStringName(ieStrRef stridx, unsigned char color, const Scriptable *speaker, STRING_FLAGS flags) const;
+	void DisplayStringName(String text, unsigned char color, const Scriptable *speaker) const;
+	void DisplayStringName(String text, const Color &color, const Scriptable *speaker) const;
+	void DisplayStringName(ieStrRef stridx, const Color &color, const Scriptable *speaker, STRING_FLAGS flags) const;
+	/** displays a string constant in the textarea, starting with speaker's name, also replaces one numeric value (it is a format string) */
+	void DisplayConstantStringNameValue(size_t stridx, unsigned char color, const Scriptable *speaker, int value) const;
+	/** displays actor name - action : parameter */
+	void DisplayConstantStringNameString(size_t stridx, unsigned char color, size_t stridx2, const Scriptable *actor) const;
 	/** displays a string in the textarea */
 	void DisplayString(const String& text) const;
+	void DisplayString(ieStrRef stridx, unsigned char color, STRING_FLAGS flags) const;
```

Comment:
I think if you resolve the strref in GemRB_DisplayString, this one can go. Same perhaps for the variant of DisplayStringName that it is using.

Reply:
Don't know in the calls from python to c++ how really works, if I change the colors will call the other function ? Or should I change the conversion also ?

Reply:
Just call GetString on the strref, so a String is passed instead. This is all on the c++ side of things.

Reply:
If I change this to send a String:

```
static PyObject* GemRB_DisplayString(PyObject * /*self*/, PyObject* args)
{
	ieStrRef strref = ieStrRef::INVALID;
	PyObject* pycol;
	int globalID = 0;
	PARSE_ARGS( args,  "iO|i", &strref, &pycol, &globalID);
	String text = core->GetString(strref, STRING_FLAGS::RESOLVE_TAGS);
	if (globalID) {
		GET_GAME();
		GET_ACTOR_GLOBAL();

		displaymsg->DisplayStringName(text, ColorFromPy(pycol), actor, STRING_FLAGS::SOUND);
	} else {
		displaymsg->DisplayString(text, ColorFromPy(pycol), STRING_FLAGS::SOUND);
	}
	Py_RETURN_NONE;
}
```
The problem will be that I have to create a new function on the message that receive the flags instead of the Scriptable target. Using the **DisplayStringName** will need a speaker to be passed.

Reply:
I see what you mean, but also that `DisplayString(ieStrRef stridx, const Color &color, STRING_FLAGS flags)` is used there, not the one I commented on.

Reply:
Yes, but that one I made private because nobody use it. So I have to make public again :
`void DisplayString(String text, const Color &color, Scriptable *target) const;`
And also do the flags logic that is on the class move it to **GUIScript.cpp** is that the wanted result ?